commands/parse/parse.go

```diff
@@ -18,40 +17,55 @@ import (
 //This command can be used for printing structured inputs from unstructured configuration inputs.
 //Can be used with '--input' or '-i' flag.
 func NewParseCommand(osExit func(int)) *cobra.Command {
-	ctx := context.Background()
-	cmd := &cobra.Command{
-		Use:     "parse <filename>",
-		Short:   "Print out structured data from your input file",
-		PreRun: func(cmd *cobra.Command, args []string) {
+	cmd := cobra.Command{
+		Use:   "parse <filename>",
+		Short: "Print out structured data from your input file",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
 			err := viper.BindPFlag("input", cmd.Flags().Lookup("input"))
 			if err != nil {
-				log.G(ctx).Fatal("Failed to bind argument:", err)
+				return fmt.Errorf("failed to bind argument: %w", err)
 			}
+			return nil
 		},
-		Run: func(cmd *cobra.Command, fileList []string) {
-			configurations, err := test.GetConfigurations(ctx, fileList)
+		RunE: func(cmd *cobra.Command, fileList []string) error {
+			out, err := parseInput(fileList)
 			if err != nil {
-				log.G(ctx).Fatalf("Your inputs could not be parsed : %s", err)
-			}
-
-			for filename, config := range configurations {
-				var prettyJSON bytes.Buffer
-				os.Stdout.WriteString((filename) + "\n")
-				out, err := json.Marshal(config)
-				if err != nil {
-					log.G(ctx).Fatalf("Error converting config to JSON out: %s", err)
-				}
-				err = json.Indent(&prettyJSON, out, "", "\t")
-				if err != nil {
-					log.G(ctx).Fatalf("Error indenting JSON output: %s", err)
-				}
-				os.Stdout.Write(prettyJSON.Bytes())
+				return fmt.Errorf("failed during parser process: %w", err)
 			}
-
-			osExit(0)
+			print(out)
+			return nil
 		},
 	}
 
 	cmd.Flags().StringP("input", "i", "", fmt.Sprintf("input type for given source, especially useful when using conftest with stdin, valid options are: %s", parser.ValidInputs()))
-	return cmd
+	return &cmd
+}
+
+func parseInput(fileList []string) ([]byte, error) {
+	ctx := context.Background()
+	configurations, err := test.GetConfigurations(ctx, fileList)
+	var bundle []byte
+	if err != nil {
+		return nil, fmt.Errorf("Your inputs could not be parsed : %w", err)
+	}
+	for filename, config := range configurations {
+		filename = filename + "\n"
+		var prettyJSON bytes.Buffer
+		out, err := json.Marshal(config)
+		if err != nil {
+			return nil, fmt.Errorf("Error converting config to JSON out: %s", err)
+		}
+		err = json.Indent(&prettyJSON, out, "", "\t")
+		if err != nil {
+			return nil, fmt.Errorf("Error indenting JSON output: %s", err)
+		}
+		bundle = append(bundle, filename...)
+		bundle = append(bundle, prettyJSON.Bytes()...)
+	}
+	return bundle, nil
+}
+
+//Will be replaced with OutputManager
```

Comment:
Can we not just use `OutputManager` now? A lot of the code here is essentially what the `jsonOutputmanager` `Flush()` function is doing.

To use the OutputManager in `parse` can be similar to context in that you just say

`out := test.GetOutputManager()`

Reply:
We are not using `OutputManager` there. If you say remove the comment, I can.

Reply:
I was just saying rather than copy and pasting the code from `OutputManager`, we could just use the OutputManager instead. That should be the preferred way to print results to the console.

Reply:
ok changing this with `OutputManager`

Reply:
@jpreese, first of all, I'm not copy-pasting from anywhere to my code. 
- Secondly, I tried to use `outputManager()` however, a few things happen when I do that.
`outputManager` has a strict structure.  
```json
[
        {
                "filename": "", // string
                "Warnings": [],
                "Failures": [],
                "Successes": [] // Error
        }
]
```
With this kind of structure, I have to even marshal the result from the `parsers` to `JSON` and cast it to `error`, like:
`str := fmt.Errorf("%s", marshalled Json)`
`err = stdout.Put(filename, test.CheckResult{Successes: []error{str}})` which seems ridiculous. 
Because `Put()` function bundles `fileName` with `result string` in a struct.
and `Flush()` Marshals them to JSON and prints.

Actually, marshaling in here is for casting the result that comes from `parsers` to `JSON` again for understanding the input structure, not for copy-paste.

- Thirdly, do all use cases have to fit the structure defined in OutputManager? Is there a rule like this?

Reply:
Currently they do not, no. I'm wondering now if `OutputManager` is just poorly named and/or the `parse` command needs its own outputmanager for a lack of better term.

We can explore different approaches as we work with the code base. If you're happy with how things are currently, I'm going to approve and feel free to merge at your convenience. Thanks!

Reply:
For me, `OutputManager` if we going to use it as default `print out` method, we have to add additional structs to get it more flexible. Addition to them, need to remove viper controls inside, 
allow it with using external parameters